src/data/polynomial/degree/erase_leading.lean

```diff
@@ -0,0 +1,166 @@
+/-
+Copyright (c) 2020 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import data.polynomial.degree.basic
+import data.polynomial.degree.to_basic
+import data.polynomial.degree.to_trailing_degree
+import data.polynomial.degree.trailing_degree
+
+open polynomial finsupp finset
+
+namespace polynomial
+
+variables {R : Type*} [semiring R] {f : polynomial R}
+
+namespace erase_leading
+
+/-- erase_lead of a polynomial f is the polynomial obtained by
+subtracting from f the leading term of f. -/
+def erase_lead (f : polynomial R) : polynomial R := ⟨ f.support \ singleton f.nat_degree , λ a : ℕ , ite (a = f.nat_degree) 0 f.coeff a , λ a , begin
+  simp only [mem_sdiff, mem_support_iff, ne.def, mem_singleton],
+  split_ifs,
+    { simp only [not_and, pi.zero_apply, not_not, eq_self_iff_true, not_true, iff_false],
+      intros a_1,
+      assumption, },
+    { split,
+        { rw and_imp,
+          intros a1 a2,
+          assumption, },
+        { intros a1,
+          exact ⟨ a1 , h ⟩, }, },
+end ⟩
+
+lemma support (f : polynomial R) :
+ (erase_lead f).support = f.support \ singleton f.nat_degree :=
+rfl
+
+@[simp] lemma coeff_nat_degree : (erase_lead f).coeff f.nat_degree = 0 :=
+begin
+  unfold erase_lead,
+  simp only [coeff_mk, if_true, pi.zero_apply, eq_self_iff_true],
+end
+
+@[simp] lemma coeff_eq_coeff_of_ne {a : ℕ} (h : a ≠ f.nat_degree) : (erase_lead f).coeff a = f.coeff a :=
+begin
+  unfold erase_lead,
+  rw coeff_mk,
+  split_ifs,
+    { exfalso,
+      solve_by_elim, },
+    { refl, },
+end
+
+lemma sum_leading_C_mul_X_pow (f : polynomial R) : f = (erase_lead f) + (C f.leading_coeff) * X^f.nat_degree :=
```

Comment:
Could you also add the subtraction version of this for when `R` is a `ring`, maybe as a `simp` lemma? I feel like when you're working with a `ring`, that's the default way to think about this function.

Reply:
I added the lemma:

@[simp] lemma sum_leading_C_mul_X_pow_ring {S : Type*} [ring S] (g : polynomial S)
 : g.erase_lead = g - (C g.leading_coeff) * X^g.nat_degree :=
eq_sub_iff_add_eq.mpr (sum_leading_C_mul_X_pow g).symm